dataprofiler/tests/profilers/test_datetime_column_profile.py

```diff
@@ -500,3 +500,35 @@ def test_json_encode_after_update(self):
         )
 
         self.assertEqual(serialized, expected)
+
+    def test_json_decode(self):
+        fake_profile_name = None
+        expected_profile = DateTimeColumn(fake_profile_name)
+
+        serialized = json.dumps(expected_profile, cls=ProfileEncoder)
+        deserialized = load_column_profile(json.loads(serialized))
+
+        test_utils.assert_profiles_equal(deserialized, expected_profile)
+
+    def test_json_decode_after_update(self):
+        fake_profile_name = "Fake profile name"
+
+        data = [2.5, 12.5, "2013-03-10 15:43:30", 5, "03/10/13 15:43", "Mar 11, 2013"]
+        df = pd.Series(data)
+
+        expected_profile = DateTimeColumn(fake_profile_name)
+        expected_profile.update(df)
+
+        serialized = json.dumps(expected_profile, cls=ProfileEncoder)
+        deserialized = load_column_profile(json.loads(serialized))
+
+        test_utils.assert_profiles_equal(deserialized, expected_profile)
+
+        data_new = ["2012-02-10T15:43:30", "20120210T154300", "March 12, 2014"]
+        df_new = pd.Series(data_new)
+
+        # validating update after deserialization
+        deserialized.update(df_new)
+
+        assert deserialized._dt_obj_min == pd.Timestamp("2012-02-10 15:43:00")
+        assert deserialized._dt_obj_max == pd.Timestamp("2014-03-12 00:00:00")
```

Comment:
Let's also make sure the new formats are added to the list here: `self.date_formats`

Reply:
added!